source/gameengine/Ketsji/KX_Scene.cpp

```diff
@@ -1691,13 +1729,6 @@ void KX_Scene::ReplaceMesh(KX_GameObject *gameobj,
     DEG_id_tag_update(&gameobj->GetBlenderObject()->id, ID_RECALC_GEOMETRY);
```

Comment:
When you use several cubes and mix "only gfx" and "only phys" depend on the cube the it is necessary to modifiy the tag_update function and move it to the end:

```
  if (use_gfx || use_phys) {
    DEG_id_tag_update(&gameobj->GetBlenderObject()->id, ID_RECALC_GEOMETRY);
  }
```